### DISCUSSION TITLE
Can I Set the Current Active Partition in the Lua API?

### DISCUSSION THREAD

**Discussion by zicklag** [jcorporation/mympd-scripts#32] [Q&A] [answered]
Is there a way through the Lua API to set the current, active partition? In the [API docs](https://jcorporation.github.io/myMPD/060-references/api/methods/) I see commands to list, create, move outputs, remove, and change settings on partitions, but I don't see one to switch the current partition.

I need to be able to create new partitions, and then move outputs to different partitions, but I don't think I can do that without being able to switch the current partition since the output move command moves outputs to the current partition.

**Comment by jcorporation** [comment, marked as answer]
With current stable version you can use following snippet to execute an API method for an other partition.

```lua
local rc, raw_result = mympd_api(partition, method, json.encode(params))
local result = json.decode(raw_result)
```

With above patch for v19.0.0 there is a new lua interface method introduced: `mympd.api_partition`

**Reply by zicklag** [reply]
Awesome, thank you very much!
